MonoGame.Framework/Input/KeyboardUtil.OpenTK.cs

```diff
@@ -1,343 +1,352 @@
-using System;
+﻿using System;
+using System.Collections.Generic;
 
 namespace Microsoft.Xna.Framework.Input
 {
-	internal static class KeyboardUtil
-	{
-		// TODO make a dictionary for fast mapping
-        public static Keys ToXna(OpenTK.Input.Key key)
+    internal static class KeyboardUtil
+    {
+        static Dictionary<OpenTK.Input.Key, Keys> _map;
+
+        static KeyboardUtil ()
+        {
+            _map = new Dictionary<OpenTK.Input.Key, Keys> ();
+            var values = Enum.GetValues (typeof(OpenTK.Input.Key));
+
+            foreach (OpenTK.Input.Key tkKey in values)
+                _map [tkKey] = _ToXna (tkKey);
+        }
+
+        static Keys _ToXna (OpenTK.Input.Key key)
         {
             switch (key)
             {
                 case OpenTK.Input.Key.A:
                     return Keys.A;
-                    
+
                 case OpenTK.Input.Key.AltLeft:
                     return Keys.LeftAlt;
-                    
+
                 case OpenTK.Input.Key.AltRight:
                     return Keys.RightAlt;
-                    
+
                 case OpenTK.Input.Key.B:
                     return Keys.B;
-                    
+
                 case OpenTK.Input.Key.Back:
                     return Keys.Back;
-                    
+
                 case OpenTK.Input.Key.BackSlash:
-                    // Issue 1012: XNA 4.0 returns this key as OemPipe, not OemBackslash,
-                    // even though OemPipe is the shifted version of the key.
-                    return Keys.OemPipe;
-                    
```

Comment:
So I've done some testing with opentk and OpenTK.Backslash key is OemPipe, but there is no key for OemBackslash, I'll probably have to submit a PR to OpenTK for that functionality.